core/src/main/java/com/github/jsonldjava/core/DocumentLoader.java

```diff
@@ -118,54 +116,55 @@ public InputStream openStreamFromURL(java.net.URL url) throws IOException {
         // or whatever is available
         request.addHeader("Accept", ACCEPT_HEADER);
 
-        final HttpResponse response = getHttpClient().execute(request);
+        final CloseableHttpResponse response = getHttpClient().execute(request);
         final int status = response.getStatusLine().getStatusCode();
         if (status != 200 && status != 203) {
+            response.close();
             throw new IOException("Can't retrieve " + url + ", status code: " + status);
         }
         return response.getEntity().getContent();
     }
 
-    protected static HttpClient getDefaultHttpClient() {
-        HttpClient result = defaultHttpClient;
+    protected static CloseableHttpClient getDefaultHttpClient() {
+        CloseableHttpClient result = defaultHttpClient;
         if (result == null) {
             synchronized (DocumentLoader.class) {
                 result = defaultHttpClient;
                 if (result == null) {
-                    // Uses Apache SystemDefaultHttpClient rather than
-                    // DefaultHttpClient, thus the normal proxy settings for the
-                    // JVM will be used
-
-                    final DefaultHttpClient client = new SystemDefaultHttpClient();
-                    // Support compressed data
-                    // http://hc.apache.org/httpcomponents-client-ga/tutorial/html/httpagent.html#d5e1238
-                    client.addRequestInterceptor(new RequestAcceptEncoding());
-                    client.addResponseInterceptor(new ResponseContentEncoding());
-                    final CacheConfig cacheConfig = new CacheConfig();
-                    cacheConfig.setMaxObjectSize(1024 * 128); // 128 kB
-                    cacheConfig.setMaxCacheEntries(1000);
-                    // and allow caching
-                    final CachingHttpClient cachingClient = new CachingHttpClient(client,
-                            cacheConfig);
-
-                    // Wrap again with JAR cache
-                    final JarCacheStorage jarCache = new JarCacheStorage();
-                    result = defaultHttpClient = new CachingHttpClient(cachingClient, jarCache,
```

Comment:
The key to fixing this was that this line created a chained CachingHttpClient set by inheriting from the underlying client.

This doesn't appear to be possible using the Builder patterns so I had to hack around it by creating layered HttpCacheStorage objects instead.